Remove FortAwesome dependencies

labkey-alan opened the pull request:
#### Rationale
This PR removes all FortAwesome dependencies and converts all usages of `<FontAwesomeIcon />` to `<span />` with font awesome class names.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/1011
* https://github.com/LabKey/biologics/pull/1705
* https://github.com/LabKey/sampleManagement/pull/1363
* https://github.com/LabKey/labbook/pull/314
* https://github.com/LabKey/inventory/pull/597
* https://github.com/LabKey/provenance/pull/130
* https://github.com/LabKey/platform/pull/3820

#### Changes
* Remove fortawesome dependencies
    * Replaced all usages of FontAwesomeIcon with spans that use the appropriate classes
* Remove CollapsiblePanel component
* Convert some components to FC
* Refactor DomainRowWarning

labkey-alan committed:
Remove FortAwesome dependencies

cnathe commented on packages/components/src/index.ts:
```diff
@@ -1281,7 +1280,6 @@ export {
     SVGIcon,
     // general components
     Alert,
-    CollapsiblePanel,
```
it looks like there are 2 usages of this in the provenance module. maybe this component should just move there? thoughts?

cnathe replied:
nvm, just saw the PR for provenance!

cnathe commented on packages/components/src/internal/components/base/DragDropHandle.tsx:
```diff
@@ -10,14 +8,15 @@ interface Props {
-            highlighted !== undefined
-                ? classNames({
-                      'field-highlighted': highlighted,
-                      'field-not-highlighted': !highlighted,
```
was `field-not-highlighted` just used to apply the gray color? I assume that isn't needed anymore

labkey-alan replied:
Yeah, it just enforces the default look, it's not needed

cnathe commented on packages/components/src/internal/components/base/LabelHelpTip.tsx:
```diff
@@ -44,7 +37,5 @@ export const LabelHelpTip: FC<Props> = memo(props => {
 LabelHelpTip.displayName = 'LabelHelpTip';
 
 LabelHelpTip.defaultProps = {
-    customStyle: {},
```
I'm curious what the usages were that were passing in `customStyle` and why? was it just to change the color or something like that?

cnathe replied:
CollapsiblePanelHeader is the only one I can find

labkey-alan replied:
Yes, and with the styles it was passing it looked really bad, alignment was way off. By removing the custom style it looks much better.

cnathe reviewed (approved):
Excited to get more dependencies removed!

labkey-alan committed:
Update test snapshots

Fix ConditionalFormatOptions tests

remove snapshots, they are not necessary we are already testing enough

Update test snapshots

Fix issues around Conditional formatting and Validation

- ValidatorModal modals now live with the underlying components
- No longer export ValidatorModal
- Fix pencil icon

Bump components

Update snapshots

Update snapshots

Bump components

labkey-alan committed:
Prep for release
